packages/core/src/utils/sorter/DropLocationDeterminer.ts

```diff
@@ -382,16 +372,22 @@ export class DropLocationDeterminer<T, NodeType extends SortableTreeNode<T>> ext
       targetNode: lastTargetNode,
       hoveredNode: lastHoveredNode,
       hoveredIndex: lastHoveredIndex,
+      mouseEvent: lastMouseEvent,
     } = this.lastMoveData;
 
     const sameHoveredNode = targetNode.equals(lastHoveredNode);
     targetNode.nodeDimensions = sameHoveredNode ? lastHoveredNode!.nodeDimensions! : this.getDim(targetNode.element!);
     const hoverIndex = this.getIndexInParent(targetNode, targetNode.nodeDimensions!, mouseX, mouseY);
     const sameHoveredIndex = hoverIndex === lastHoveredIndex;
-    const sameHoverPosition = sameHoveredNode && sameHoveredIndex;
+    const isWithinDropArea = targetNode.isWithinDropBounds(mouseX, mouseY);
+    const sameHoverPosition =
+      sameHoveredNode &&
+      sameHoveredIndex &&
+      isWithinDropArea === targetNode.isWithinDropBounds(lastMouseEvent?.clientX ?? 0, lastMouseEvent?.clientY ?? 0);
```

Comment:
Here, if last movement was over an undraggable area but now it's on draggable area, don't use the cached values